Makd.mak

```diff
@@ -565,6 +565,40 @@ $O/test-%.stamp: $O/test-%
 	$(call exec,$< $(ITFLAGS),$<,run)
 	$Vtouch $@
 
+# examples rules
+##########################
+
+# examples are assumed to be standalone programs, so we recursively search
+# for files examples/*.d and assume that each file is a separate example
+# program. Files can exist in sub-directories under examples/ dir.
+# The sources list is filtered through the $(EXAMPLES_FILTER_OUT) variable 
+# contents (using the Make function $(filter-out)), so you can exclude an
+# example by adding the example file path (as an absolute path using $C) to
+# this variable.
+
+# The 'example' target builds all the examples.
+.PHONY: example
+example: $(patsubst $T/example/%.d,$O/example/%,\
+	$(filter-out $(EXAMPLE_FILTER_OUT),$(shell find $T/example/ -name "*.d")))
+
+# General rule to build an example program.
+$O/example/%: BUILD.d.depfile = $O/example/$*.mak
```

Comment:
You should be able to infer it from both the `integrationtest` and `$B/%:` rules. I hope that's enlightening, because to go further I would have to go to the code myself and have a look.